### PR TITLE
Automation: Inject RuleManager instead of non existing ModuleHandlerCallback

### LINKED ISSUE #5976: 'smarthome:automation listRules' causes an exception
OH 2.4 snapshot 1319 (ESH 0.10.0.201807241640) (this does not occur on snapshot 1304)

I am using the Experimental Rules Engine (JSR223). Running this in Karaf results in an exception...
```
openhab> smarthome:automation listRules
An unexpected error occurred during execution.
```
The openhab.log shows this...
```
2018-07-28 00:24:22.471 [ERROR] [rg.eclipse.smarthome.io.console.ConsoleInterpreter] - An error occurred while executing the console command.
java.lang.NullPointerException: null
        at org.eclipse.smarthome.automation.internal.commands.Printer.printRules(Printer.java:115) [210:org.eclipse.smarthome.automation.commands:0.10.0.201807241640]
        at org.eclipse.smarthome.automation.internal.commands.AutomationCommandList.listRules(AutomationCommandList.java:178) [210:org.eclipse.smarthome.automation.commands:0.10.0.201807241640]
        at org.eclipse.smarthome.automation.internal.commands.AutomationCommandList.execute(AutomationCommandList.java:85) [210:org.eclipse.smarthome.automation.commands:0.10.0.201807241640]
        at org.eclipse.smarthome.automation.internal.commands.AutomationCommands.executeCommand(AutomationCommands.java:517) [210:org.eclipse.smarthome.automation.commands:0.10.0.201807241640]
        at org.eclipse.smarthome.automation.internal.commands.AutomationCommandsPluggable.execute(AutomationCommandsPluggable.java:184) [210:org.eclipse.smarthome.automation.commands:0.10.0.201807241640]
        at org.eclipse.smarthome.io.console.ConsoleInterpreter.execute(ConsoleInterpreter.java:56) [111:org.eclipse.smarthome.io.console:0.10.0.201807241640]
        at org.eclipse.smarthome.io.console.karaf.internal.CommandWrapper.execute(CommandWrapper.java:79) [112:org.eclipse.smarthome.io.console.karaf:0.10.0.201807241640]
        at org.apache.karaf.shell.impl.console.osgi.secured.SecuredCommand.execute(SecuredCommand.java:68) [12:org.apache.karaf.shell.core:4.1.5]
        at org.apache.karaf.shell.impl.console.osgi.secured.SecuredCommand.execute(SecuredCommand.java:86) [12:org.apache.karaf.shell.core:4.1.5]
        at org.apache.felix.gogo.runtime.Closure.executeCmd(Closure.java:571) [12:org.apache.karaf.shell.core:4.1.5]
        at org.apache.felix.gogo.runtime.Closure.executeStatement(Closure.java:497) [12:org.apache.karaf.shell.core:4.1.5]
        at org.apache.felix.gogo.runtime.Closure.execute(Closure.java:386) [12:org.apache.karaf.shell.core:4.1.5]
        at org.apache.felix.gogo.runtime.Pipe.doCall(Pipe.java:417) [12:org.apache.karaf.shell.core:4.1.5]
        at org.apache.felix.gogo.runtime.Pipe.call(Pipe.java:229) [12:org.apache.karaf.shell.core:4.1.5]
        at org.apache.felix.gogo.runtime.Pipe.call(Pipe.java:59) [12:org.apache.karaf.shell.core:4.1.5]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```

### PR DESCRIPTION
**Pull request by triller-telekom**
Fixes #5976

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java [hunk 1]
```diff
@@ -142,13 +146,13 @@ protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegist
     }
 
     /**
-     * Bind the {@link ModuleHandlerCallback} service - called from DS.
+     * Bind the {@link RuleEngine} service - called from DS.
      *
      * @param ruleEngine RuleManager service.
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java]: ruleEngine -> ruleManager

File: bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java [hunk 2]
```diff
@@ -142,13 +146,13 @@ protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegist
     }
 
     /**
-     * Bind the {@link ModuleHandlerCallback} service - called from DS.
+     * Bind the {@link RuleEngine} service - called from DS.
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java]: RuleEngine -> RuleManager

File: bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java [hunk 3]
```diff
@@ -142,13 +146,13 @@ protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegist
...
      *
      * @param ruleEngine RuleManager service.
      */
     @Reference
-    protected void setRuleEngine(ModuleHandlerCallback ruleEngine) {
-        this.callback = ruleEngine;
+    protected void setRuleManager(RuleManager ruleManager) {
+        this.callback = ruleManager;
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java]: probably also worth to rename `callback` to `ruleManager`.

File: bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java [hunk 4]
```diff
@@ -43,7 +45,7 @@
  * @author Ana Dimova - Initial Contribution
  * @author Kai Kreuzer - refactored (managed) provider and registry implementation
  */
-@Component(immediate = true)
+@Component(immediate = true, name = "org.eclipse.smarthome.automation.commands")
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java]: Do we require a specific name here?

Comment (triller-telekom) [reply bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java]: It was like that before in the xml file, which is why I kept it.

Comment (triller-telekom) [reply bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java]: Just tested it: Works without a specific name.

**Comment by triller-telekom** [comment]
Comments are addressed

File: bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java [hunk 5]
```diff
@@ -408,8 +412,8 @@ protected String buildCommandUsage(final String syntax, final String description
 
     @Override
     public RuleStatus getRuleStatus(String ruleUID) {
-        if (callback != null) {
-            RuleStatusInfo rsi = callback.getStatusInfo(ruleUID);
+        if (ruleManager != null) {
```
Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java]: The rule manager is a mandatory reference, so it cannot be null if the component is active.

File: bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java [hunk 6]
```diff
@@ -418,8 +422,8 @@ public RuleStatus getRuleStatus(String ruleUID) {
 
     @Override
     public void setEnabled(String uid, boolean isEnabled) {
-        if (callback != null) {
-            callback.setEnabled(uid, isEnabled);
+        if (ruleManager != null) {
```
Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java]: See above (lifecycle)

**Comment by triller-telekom** [comment]
You are right, the `AutomationCommandsPluggable` does not have to be activated immediately. I have removed the immediate activation. I also removed the `null` checks for the mandatory reference.

**Pull request merged**